platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java

```diff
@@ -684,10 +699,9 @@ public void setUseKey(Boolean useKey) {
         @Override
         public boolean doCredentialsMatch(AuthenticationToken token, AuthenticationInfo info) {
             if (token instanceof SAMLAuthenticationToken) {
-                SecurityToken oldToken = (SecurityToken) token.getCredentials();
-                SecurityToken newToken = (SecurityToken) info.getCredentials();
-                return oldToken.getId()
-                        .equals(newToken.getId());
+                Object oldToken = token.getCredentials();
+                Object newToken = info.getCredentials();
+                return oldToken.equals(newToken);
```

Comment:
Yeah, it really is the same object for the renewal case, which is what this section deals with. The renewal itself was constantly throwing an exception due to the fact that multiple requests got there at the same exact time. This particular code was throwing a null pointer exception once I fixed the concurrent requests.